src/firmware/riscv/wormhole/l1_address_map.h

```diff
@@ -18,120 +18,120 @@ struct mailbox_type {
-  static constexpr std::int32_t UNPACK_PACK_PERF_BUF_SIZE_LEVEL_0 = 640; // smaller buffer size for limited logging
-  static constexpr std::int32_t UNPACK_PACK_PERF_BUF_SIZE_LEVEL_1 = (12 * 1024 - 768)/2 - MATH_PERF_BUF_SIZE/2 - (PERF_QUEUE_HEADER_SIZE + PERF_RISC_MAILBOX_SIZE + PERF_UNUSED_SIZE)/2 - BRISC_PERF_BUF_SIZE/2;
-  
-  static constexpr std::int32_t PERF_QUEUE_HEADER_ADDR = FIRMWARE_BASE + BRISC_FIRMWARE_SIZE + ZEROS_SIZE;
-  static constexpr std::int32_t PERF_RISC_MAILBOX_ADDR = PERF_QUEUE_HEADER_ADDR + PERF_QUEUE_HEADER_SIZE;
-  static constexpr std::int32_t BRISC_PERF_BUF_BASE_ADDR = PERF_RISC_MAILBOX_SIZE + PERF_UNUSED_SIZE + PERF_RISC_MAILBOX_ADDR;
-  static constexpr std::int32_t MATH_PERF_BUF_BASE_ADDR = BRISC_PERF_BUF_BASE_ADDR + BRISC_PERF_BUF_SIZE;
-  static constexpr std::int32_t UNPACK_PACK_PERF_BUF_BASE_ADDR = MATH_PERF_BUF_BASE_ADDR + MATH_PERF_BUF_SIZE;
-  static constexpr std::int32_t PERF_NUM_THREADS = 5;
-
-  static constexpr std::int32_t PERF_QUEUE_PTRS = PERF_QUEUE_HEADER_ADDR;
-  static constexpr std::int32_t PERF_THREAD_HEADER = PERF_QUEUE_PTRS + 8;
-  static constexpr std::int32_t PERF_WR_PTR_COPY = PERF_QUEUE_PTRS + 12;
-
-  static constexpr std::int32_t WALL_CLOCK_L = 0xFFB121F0;
-  static constexpr std::int32_t WALL_CLOCK_H = 0xFFB121F8;
+  static constexpr uint32_t PERF_BUF_SIZE = FIRMWARE_SIZE - BRISC_FIRMWARE_SIZE - ZEROS_SIZE;
+
+  static constexpr uint32_t PERF_QUEUE_HEADER_SIZE = 16;
+  static constexpr uint32_t PERF_RISC_MAILBOX_SIZE = 16;
+  static constexpr uint32_t PERF_UNUSED_SIZE = 32;
+
+  static constexpr uint32_t MATH_PERF_BUF_SIZE = 64;
+  static constexpr uint32_t BRISC_PERF_BUF_SIZE = 640; // Half of this value must be 32B aligned
+  static constexpr uint32_t UNPACK_PACK_PERF_BUF_SIZE_LEVEL_0 = 640; // smaller buffer size for limited logging
+  static constexpr uint32_t UNPACK_PACK_PERF_BUF_SIZE_LEVEL_1 = (12 * 1024 - 768)/2 - MATH_PERF_BUF_SIZE/2 - (PERF_QUEUE_HEADER_SIZE + PERF_RISC_MAILBOX_SIZE + PERF_UNUSED_SIZE)/2 - BRISC_PERF_BUF_SIZE/2;
+
+  static constexpr uint32_t PERF_QUEUE_HEADER_ADDR = FIRMWARE_BASE + BRISC_FIRMWARE_SIZE + ZEROS_SIZE;
+  static constexpr uint32_t PERF_RISC_MAILBOX_ADDR = PERF_QUEUE_HEADER_ADDR + PERF_QUEUE_HEADER_SIZE;
+  static constexpr uint32_t BRISC_PERF_BUF_BASE_ADDR = PERF_RISC_MAILBOX_SIZE + PERF_UNUSED_SIZE + PERF_RISC_MAILBOX_ADDR;
+  static constexpr uint32_t MATH_PERF_BUF_BASE_ADDR = BRISC_PERF_BUF_BASE_ADDR + BRISC_PERF_BUF_SIZE;
+  static constexpr uint32_t UNPACK_PACK_PERF_BUF_BASE_ADDR = MATH_PERF_BUF_BASE_ADDR + MATH_PERF_BUF_SIZE;
+  static constexpr uint32_t PERF_NUM_THREADS = 5;
+
+  static constexpr uint32_t PERF_QUEUE_PTRS = PERF_QUEUE_HEADER_ADDR;
+  static constexpr uint32_t PERF_THREAD_HEADER = PERF_QUEUE_PTRS + 8;
+  static constexpr uint32_t PERF_WR_PTR_COPY = PERF_QUEUE_PTRS + 12;
+
+  static constexpr uint32_t WALL_CLOCK_L = 0xFFB121F0;
```

Comment:
How did dhis work previously? This would result in a negative value, right?

Reply:
UMD used to get away with this brokenness because it would do something like,

```
void b(uint32_t i) { /* fixed: truncated back to 32 bits */ }
void a(uint64_t i) { /* oops: now it's sign extended */ b(i); }

int32_t WALL_CLOCK_L = 0xFFB121F0; /* yes, this is negative */
a(WALL_CLOCK_L);
```
We had some bugs when the equivalent of `b(uint32_t)` was changed to uint64.

Reply:
I think this was fixed by changing Metal's version of the address maps to be unsigned.